src/transformStateWithClones.js

```diff
@@ -7,7 +7,47 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let currentState = { ...state };
+
+  const resultArray = [];
+
+  for (const action of actions) {
+    let newState = { ...currentState };
+
+    switch (action.type) {
+      case 'addProperties':
+
+        if (action.extraData) {
+          Object.assign(newState, action.extraData);
+        }
+        break;
+
+      case 'removeProperties':
```

Comment:
Consider will be better if you create the variable for this word and use it here